doc(archive_maintenance):  arch contract data maintenance description (#1062)
bonitasoft/bonita-doc: md/maintenance-operation.md

# Bonita Platform Maintenance Operation

> We learn how we could execute maintenance operations to reduce disk space using by `Bonita` database.

Reviewer: ```suggestion
Learn how maintenance operations can help reducing disk space used by the Bonita Platform.
```

> # Archive Contract Data 

Reviewer: ```suggestion
# Purge archive tables
Archive tables contain the execution history of the Bonita Platform. Some of these tables can be purged to reduce size on disk.

## Archive Contract Data
```

> Archive contract data table keeps the information of all contracts send to execute tasks or instantiate processes. After of lot of tasks or process execution,  the `archive_contract_data` table could become very big and consume a lot of disk spaces.

Reviewer: ```suggestion
Archive contract data table keeps the information of all contracts sent to execute tasks or instantiate processes. After of a lot of tasks or processes were executed, the `archive_contract_data` table could become very big and consume a lot of disk spaces.
```

Those information are not required by Bonita Platform to work so table could be purge to reduce disk space.
                                                                                                                                 

> ## Overview 

Reviewer: ```suggestion
```

These section will describe how to purge the `archive_contract_data` table, depending on the version of Bonita Platform installed.

> **Warning**

Reviewer: ```suggestion
::: warning
```

> _These operation is destructive, data's will be not usable for audit or analytic process. It is highly recommended to backup data before purge._

Reviewer: ```suggestion
_This operation is destructive, data will be not usable for audit or analytic purpose. It is highly recommended to backup data before the purge._
:::
```

_Keep in mind that the contract data's will continue to be archived and the table will continue to be filled,  even after purge has been done. Hoverer, it's possible to deactivate the archiving of `contract data's`, please see ([Configurable Archive](configurable-archive.md).)_  
   

> ## Prior to bonita 7.7

Reviewer: ```suggestion
```

To delete all rows from the `arch_contract_data`, you can use the `TRUNCATE TABLE` statement.

~~~~
    TRUNCATE TABLE arch_contract_data;
~~~~

> After a migration to the bonita 7.7 version, the data's `archive_contract_data` table has been moved to  new `arch_contract_data_backup` table. 

Reviewer: ```suggestion
::: info
**Note**: After a migration to the bonita 7.7 version, the data of the `archive_contract_data` table has been moved to the new `arch_contract_data_backup` table for migration purpose.
In that case, If the `live-migration` tool was not executed (please see [Bonita migration steps]|migrate-from-an-earlier-version-of-bonita-bpm.md), you will also need to drop the table `arch_contract_data_backup`:
```

> In case of  the `live-migration` tool was not executed (please see [Bonita migration steps]|migrate-from-an-earlier-version-of-bonita-bpm.md), please run  `DROP TABLE` statement.

Reviewer: ```suggestion
```

   
~~~~
    DROP TABLE arch_contract_data_backup;

> ~~~~

Reviewer: ```suggestion
~~~~
:::
```

# Bonita Platform Maintenance Operation

Learn how maintenance operations can help reducing disk space used by the Bonita Platform.

# Purge archive tables
Archive tables contain the execution history of the Bonita Platform. Some of these tables can be purged to reduce size on disk.

## Archive Contract Data

Archive contract data table keeps the information of all contracts sent to execute tasks or instantiate processes. After of a lot of tasks or processes were executed, the `archive_contract_data` table could become very big and consume a lot of disk spaces.

> Those information are not required by Bonita Platform to work so table could be purge to reduce disk space.

Reviewer: ```suggestion
Those information are not required by Bonita Platform to work so table could be purge to reduce disk space usage.
```

These section will describe how to purge the `archive_contract_data` table, depending on the version of Bonita Platform installed.

::: warning

_This operation is destructive, data will be not usable for audit or analytic purpose. It is highly recommended to backup data before the purge._
:::

_Keep in mind that the contract data's will continue to be archived and the table will continue to be filled,  even after purge has been done. Hoverer, it's possible to deactivate the archiving of `contract data's`, please see ([Configurable Archive](configurable-archive.md).)_  
   

To delete all rows from the `arch_contract_data`, you can use the `TRUNCATE TABLE` statement.

~~~~
    TRUNCATE TABLE arch_contract_data;
~~~~

::: info
**Note**: After a migration to the bonita 7.7 version, the data of the `archive_contract_data` table has been moved to the new `arch_contract_data_backup` table for migration purpose.
In that case, If the `live-migration` tool was not executed (please see [Bonita migration steps]|migrate-from-an-earlier-version-of-bonita-bpm.md), you will also need to drop the table `arch_contract_data_backup`:

> In case of  the `live-migration` tool was not executed (please see [Bonita migration steps]|migrate-from-an-earlier-version-of-bonita-bpm.md), please run  `DROP TABLE` statement.

Reviewer: this statement is still duplicated
